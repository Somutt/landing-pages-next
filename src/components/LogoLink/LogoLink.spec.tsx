import { screen } from '@testing-library/react';
import { LogoLink } from '.';
import { renderTheme } from '../../utils/renderTheme';

describe('<LogoLink />', () => {
  it('should render text logo', () => {
    renderTheme(<LogoLink text='Logo' link='#target'></LogoLink>);
    const link = screen.getByRole('heading', { name: /logo/i });
    expect(link.firstChild).toHaveAttribute('href', '#target');
  });

  it('should render image logo', () => {
    renderTheme(
      <LogoLink text='Logo' imgSrc='img1/img.png' link='#target'></LogoLink>,
    );
    const link = screen.getByRole('heading', { name: /logo/i });
    expect(link.firstChild.firstChild).toHaveAttribute('src', 'img1/img.png');
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <LogoLink text='Logo' imgSrc='img1/img.png' link='#target'></LogoLink>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
