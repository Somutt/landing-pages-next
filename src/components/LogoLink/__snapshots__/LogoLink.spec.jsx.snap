// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should match snapshot 1`] = `
.c1 {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
}

.c1 >img {
  height: 2rem;
}

.c0 {
  color: #0A1128;
  font-size: 1.6rem;
  text-transform: uppercase;
}

<h1
  class="c0"
>
  <a
    class="c1"
    href="#target"
  >
    <img
      alt="Logo"
      src="img1/img.png"
    />
  </a>
</h1>
`;
